### PR TITLE
RFC: Use rkt api service to verify container

### PR DESCRIPTION
**Pull request by sjpotter**
This doesn't parse the file system but asks the rkt api service to determine if its responsible for a specific cgroup.

However, there is a slight race when systemd is used to create the initial cgroup so the "pod" cgroup can be missed.  With that said, this seems good enough for k8s hpa as those are focused on the app containers which work fine

### CODE REVIEW

**Comment by sjpotter** [comment]
also includes my other PR of the rkt bump

**Comment by sjpotter** [comment]
@timstclair

**Comment by sjpotter** [comment]
refreshed PR as one commit was accepted

File: container/rkt/client.go [hunk 1]
```diff
@@ -37,25 +39,83 @@ var (
 	once         sync.Once
 )
 
+type rktVersion struct {
+	*semver.Version
+}
+
+func (r rktVersion) Compare(other string) (int, error) {
```
Comment (timstclair) [inline container/rkt/client.go]: (optional) This is already implemented by "github.com/blang/semver".Version.Compare

File: container/rkt/client.go [hunk 2]
```diff
@@ -37,25 +39,83 @@ var (
...
 		conn, err := net.DialTimeout("tcp", defaultRktAPIServiceAddr, timeout)
 		if err != nil {
 			rktClient = nil
 			rktClientErr = fmt.Errorf("rkt: cannot tcp Dial rkt api service: %v", err)
 			return
 		}
-
 		conn.Close()
 
+		rktClient = nil
```
Comment (timstclair) [inline container/rkt/client.go]: unnecessary?

**Comment by timstclair** [comment]
I like the idea, but I'm nervous about the potential race condition. Can you elaborate more on how that can happen? Is it possible to fix it within the rkt API?

File: container/rkt/client.go (hunk 1 above)
Comment (sjpotter) [inline container/rkt/client.go]: was just mirroring what is done for rkt in k8s

File: container/rkt/client.go (hunk 2 above)
Comment (sjpotter) [inline container/rkt/client.go]: is it bad unnecessary?  I was just trying to be clear with the code.

**Comment by sjpotter** [comment]
So as of now it can't be easily fixed.  the api service doesn't launch rkt pods, it just monitors the state, so its possible for cgroups to be created (such as when they are created by systemd) before rkt has done anything and hence nothing for hte api-service to return.  

People seem to believe that once coreos ship systemd 229 (and hence will be part of the stage1) that we will be able to handle this better, but healthy skepticism till I see it.

I'd note that the old code works for cases where a user did a "rkt run" on the command line, but doesn't work when rkt is called from a systemd unit at all (ala k8s usage)

**Comment by sjpotter** [comment]
As of now, think this is going to be deprecated in favor of my other PR, just not closing it yet

**Comment by sjpotter** [comment]
other approach was used

**Pull request closed without merging**